### PR TITLE
Added check for get optional int and float value for rapidjson

### PR DESCRIPTION
**Pull request by kdiluca**
@kevinkreiser is this at all what you had in mind?  I was unable to find something better for a ptree check.

### CODE REVIEW

File: src/baldr/location.cc [hunk 1]
```diff
@@ -144,7 +144,10 @@ Location Location::FromRapidJson(const rapidjson::Value& d){
   Location location{{*lon,*lat}, stop_type};
 
   location.date_time_ = GetOptionalFromRapidJson<std::string>(d, "/date_time");
-  location.heading_ = GetOptionalFromRapidJson<float>(d, "/heading");
+
+  if (d.HasMember("heading"))
+    location.heading_ = d["heading"].IsFloat() ?  GetOptionalFromRapidJson<float>(d, "/heading").get_value_or(-1) : GetOptionalFromRapidJson<int>(d, "/heading").get_value_or(-1);
```
Comment (kevinkreiser) [inline src/baldr/location.cc]: we should not set this to `-1`

Comment (kevinkreiser) [inline src/baldr/location.cc]: you may as well just get `float` here too. and if its not `boost::none` then set `location.heading_`

File: src/loki/isochrone_action.cc [hunk 2]
```diff
@@ -31,41 +31,43 @@ namespace valhalla {
...
       size_t prev = 0;
       for(const auto& contour : *contours) {
-        const int c = GetOptionalFromRapidJson<int>(contour, "/time").get_value_or(-1);
-        if(c < prev || c == -1)
-          throw valhalla_exception_t{400, 111};
-        if(c > max_time)
-          throw valhalla_exception_t{400, 151, std::to_string(max_time)};
-        prev = c;
+        if (contour.HasMember("time")) {
+          const int c = contour["time"].IsFloat() ?  GetOptionalFromRapidJson<float>(contour, "/time").get_value_or(-1) : GetOptionalFromRapidJson<int>(contour, "/time").get_value_or(-1);
```
Comment (kevinkreiser) [inline src/loki/isochrone_action.cc]: i would just do like you were doing with boost get optional and just always get `float`, dont check for the member and dont check if its float. basically keep the code the same but change int to float in the template paramter

**Comment by kevinkreiser** [comment]
i think that there is a way to actually fix this. by that i mean that we can update the rapidjson util function, that we use to get a value, to handle this case. for boost there is no way to fix it directly so we'd need to make a similar util function to handle it. id rather just see boost ptree request parsing bite the dust completely before wasting time on that option though.

File: src/baldr/location.cc [hunk 3]
```diff
@@ -144,7 +144,9 @@ Location Location::FromRapidJson(const rapidjson::Value& d){
   Location location{{*lon,*lat}, stop_type};
 
   location.date_time_ = GetOptionalFromRapidJson<std::string>(d, "/date_time");
-  location.heading_ = GetOptionalFromRapidJson<float>(d, "/heading");
+
+  location.heading_ = GetOptionalFromRapidJson<int>(d, "/heading");
```
Comment (kevinkreiser) [inline src/baldr/location.cc]: holy whitespace batman

**Comment by kevinkreiser** [comment]
:rocket:

**Comment by kdiluca** [comment]
This fixes https://github.com/valhalla/valhalla/issues/599

**Pull request merged**